bin/init_exercise.py

```diff
@@ -0,0 +1,411 @@
+            generate_readme(name, True)
+
+
+def make_new_exercise(name, exercise_dir):
+    print("Creating new exercise from scratch...")
+    with inside(exercise_dir):
+        os.mkdir('.meta')
+        with inside('.meta'):
+            with open('description.md', 'w') as description:
+                print("Describe your exercise here", file=description)
+                print(file=description)
+                print("Don't forget that `README.md` is automatically generated; update this within `.meta\description.md`.", file=description)
+            with open('metadata.yml', 'w') as metadata:
+                print("---", file=metadata)
+                print("blurb: \"{} (created {})\"".format(
+                    name, datetime.now().isoformat()), file=metadata)
+                print("source: \"\"", file=metadata)
+                print("source_url: \"\"", file=metadata)
+        with inside('tests'):
+            with open('{}.rs'.format(name), 'a') as tests_rs:
+                print("// Add your tests here", file=tests_rs)
+
+
+def make_exercise_with_specifications(name, exercise_dir, canonical_data):
+    print("Creating exercise from specification...")
+    with open(os.path.join(exercise_dir, 'tests', '{}.rs'.format(name)), 'a') as tests_rs:
+        print("fn process_case(input: ???, expected: ???) {", file=tests_rs)
+        print("    unimplemented!()", file=tests_rs)
+        print("}", file=tests_rs)
+        print(file=tests_rs)
+        first_case = True
+
+        def literal(item):
+            if isinstance(item, str):
+                return '"{}"'.format(item)
+            elif isinstance(item, list):
+                return "vec![{}]".format(
+                    ', '.join((literal(i) for i in item))
+                )
+            elif isinstance(item, dict):
+                return """
+                    {
+                        let mut hm = ::std::collections::HashMap::new();
+                        {}
+                        hm
+                    }
+                """.strip().format('\n                        '.join((
+                    "hm.insert({}, {});".format(literal(k), literal(v))
+                    for k, v in item.items()))
+                )
+            else:
+                return str(item)
+
+        def write_case(case):
+            nonlocal first_case
+
+            print("#[test]", file=tests_rs)
+            if first_case:
+                first_case = False
+            else:
+                print("#[ignore]", file=tests_rs)
+            print("fn test_{}() {".format(to_item_name(case['description'])), file=tests_rs)
+            print("    process_case({}, {});".format(literal(case['input']), literal(case['expected'])),
+                  file=tests_rs)
+            print("}", file=tests_rs)
+            print(file=tests_rs)
+
+        def write_cases(cases):
+            for item in cases:
+                if all(key in item for key in ('description', 'input', 'expected')):

```

Comment:
Note that the input isn't necessarily in `input`, but it would be difficult to come up with something that works for all exercises. Some tracks have created ideas and you may find some of them in https://github.com/exercism/discussions/issues/155#issue-233471715 . I wouldn't mind enforcing in the problem-specifications schema that all inputs must appear under `input` (possibly as an object that contains multiple keys, since some problems have multiple inputs)

Reply:
> I wouldn't mind enforcing in the problem-specifications schema that all inputs must appear under input (possibly as an object that contains multiple keys, since some problems have multiple inputs)

but obviously that is far out of scope for this PR.

Reply:
I'm aware that some specifications include their inputs under other keys, but I think the best way forward for scripts like this is to enforce that all input is in `input` (possibly as an object) and all expected results are in `expected`.  As you say, that's an issue out of scope for this PR.

The closest I've seen to a schema for the `canonical-data.json` format is in the [problem-specifications README](https://github.com/exercism/problem-specifications/blob/master/README.md), and that uses `input` and `expected`, so that's all I feel the need to support for now.

Reply:
You're right that more often than not, `input` isn't actually used, which poses a problem. I may introduce code to do something like what vimscript does, now that you've pointed out the issue.